### ISSUE TITLE
Support for `Ed(X)25519`

### ISSUE THREAD

**Issue by olegbespalov** [grafana/k6#4263]
# What?

Even though original WebCrypto API doesn't include support of the Curve25519, but many of webcrypto implementation does https://github.com/w3c/webcrypto/pull/362

Even WebPlatfrom test suit tests this support, which led us to patch https://github.com/grafana/k6/blob/master/internal/js/modules/k6/webcrypto/tests/wpt-patches/WebCryptoAPI__generateKey__failures.js.patch

Draft can be found https://wicg.github.io/webcrypto-secure-curves/

From the Golang's perspective, support also looks good since we have packages:
* https://pkg.go.dev/crypto/ed25519@go1.23.2
* https://pkg.go.dev/golang.org/x/crypto@v0.28.0/curve25519

# Why?

Seems like support of this Curve25519 is demanded, so having it in k6 makes k6's implementation more attractive and useful for customers.

**Comment by McMastS** [comment]
Hey @olegbespalov! I'd like to take this issue on, I'll hopefully have a draft PR up by the end of the weekend.

**Comment by olegbespalov** [comment]
Hey @McMastS !

Sure, go ahead, thanks for doing that!

Just in case, I've updated the description to reflect the actual path (we've got back-merged extensions code to k6).
* https://github.com/grafana/k6/blob/master/internal/js/modules/k6/webcrypto/tests/wpt-patches/WebCryptoAPI__generateKey__failures.js.patch

Don't hesitate to ask if you need any help or want to discuss something!

Cheers!

**Comment by McMastS** [comment]
Hey @olegbespalov! I've uncommented the `X25519` and `Ed25519` lines in that test but I can't actually get the go test to fail locally. 

 My `allTestVectors` now looks like this in `internal/js/modules/k6/webcrypto/tests/wpt/WebCryptoAPI/generateKey/successes.js`:
```
    var allTestVectors = [ // Parameters that should work for generateKey
        {name: "AES-CTR",  resultType: CryptoKey, usages: ["encrypt", "decrypt", "wrapKey", "unwrapKey"], mandatoryUsages: []},
        {name: "AES-CBC",  resultType: CryptoKey, usages: ["encrypt", "decrypt", "wrapKey", "unwrapKey"], mandatoryUsages: []},
        {name: "AES-GCM",  resultType: CryptoKey, usages: ["encrypt", "decrypt", "wrapKey", "unwrapKey"], mandatoryUsages: []},
        // {name: "AES-KW",   resultType: CryptoKey, usages: ["wrapKey", "unwrapKey"], mandatoryUsages: []},
        {name: "HMAC",     resultType: CryptoKey, usages: ["sign", "verify"], mandatoryUsages: []},
        {name: "RSASSA-PKCS1-v1_5", resultType: "CryptoKeyPair", usages: ["sign", "verify"], mandatoryUsages: ["sign"]},
        {name: "RSA-PSS",  resultType: "CryptoKeyPair", usages: ["sign", "verify"], mandatoryUsages: ["sign"]},
        {name: "RSA-OAEP", resultType: "CryptoKeyPair", usages: ["encrypt", "decrypt", "wrapKey", "unwrapKey"], mandatoryUsages: ["decrypt", "unwrapKey"]},
        {name: "ECDSA",    resultType: "CryptoKeyPair", usages: ["sign", "verify"], mandatoryUsages: ["sign"]},
        {name: "ECDH",     resultType: "CryptoKeyPair", usages: ["deriveKey", "deriveBits"], mandatoryUsages: ["deriveKey", "deriveBits"]},
        {name: "Ed25519",  resultType: "CryptoKeyPair", usages: ["sign", "verify"], mandatoryUsages: ["sign", "wrapKey"]},
        // {name: "Ed448",    resultType: "CryptoKeyPair", usages: ["sign", "verify"], mandatoryUsages: ["sign"]},
        {name: "X25519",   resultType: "CryptoKeyPair", usages: ["deriveKey", "deriveBits"], mandatoryUsages: ["deriveKey", "deriveBits"]},
        // {name: "X448",     resultType: "CryptoKeyPair", usages: ["deriveKey", "deriveBits"], mandatoryUsages: ["deriveKey", "deriveBits"]},
    ];
```

The go test passes in about 20 seconds so it is running something, and if I throw an error on line 64 of that same file then the test "passes" in less than a second so it's definitely getting cut short. Is there any Sobek setup or something I'm missing to get these tests to propagate the JS error to the Go test?

**Comment by olegbespalov** [comment]
@McMastS you're absolutely right 🤔 If I uncomment these lines locally, the tests currently pass without any failures.

![Image](https://github.com/user-attachments/assets/8040caea-c7f9-49cd-af77-577d16add9b6)

But the simple script like:

```js
import { crypto } from "k6/experimental/webcrypto";

export default async function () {
  const key = await crypto.subtle.generateKey(
    {
      name: "Ed25519"
    },
    true,
    ["sign", "verify"]
  );

  console.log(JSON.stringify(key));
}
```

Fails with the:

```
ERRO[0000] Uncaught (in promise) NotSupportedError: algorithm "ED25519" doesn't support (in implementation) operation "generateKey"  executor=per-vu-iterations scenario=default
```

![Image](https://github.com/user-attachments/assets/09e6626c-7a2c-467b-afed-a8ec508a785c)

It's probably some degradation (maybe in the tests happened), but I need to check deeper, why it so, and I'm going to open a separated issue for a better visibility of the problem. 

In theory, the work on this issue still could be continued since the key generation test case is simple enough, but doesn't actually check if the functionality there. But there are tests in `internal/js/modules/k6/webcrypto/tests/wpt/WebCryptoAPI/sign_verify` with entrypoint `eddsa_curve25519.https.any.js` that could help validate implementation.

**Comment by McMastS** [comment]
Thanks for looking into it! Good to know I wasn't going crazy hahaha. I'll keep working on it without those tests and verify with manual Javascript scripts for now.
